File: 151003/Romanko/project/publisher/src/main/java/org/example/publisher/impl/user/mapper/UserMapper.java
package org.example.publisher.impl.user.mapper;

import org.example.publisher.impl.user.User;
import org.example.publisher.impl.user.dto.UserRequestTo;
import org.example.publisher.impl.user.dto.UserResponseTo;

import java.util.List;

public interface UserMapper {

    UserRequestTo userToRequestTo(User user);

    List<UserRequestTo> userToRequestTo(Iterable<User> user);

    User dtoToEntity(UserRequestTo userRequestTo);

    List<User> dtoToEntity(Iterable<UserRequestTo> userRequestTos);

    UserResponseTo userToResponseTo(User user);

    List<UserResponseTo> userToResponseTo(Iterable<User> user);
    
    
}

File: 151003/Barylko/REST/TaskREST/Mappers/UserMapper.cs
using Riok.Mapperly.Abstractions;
using TaskREST.Dto.Request.CreateTo;
using TaskREST.Dto.Request.UpdateTo;
using TaskREST.Dto.Response;
using TaskREST.Models;

namespace TaskREST.Mappers;

[Mapper]
public static partial class UserMapper
{
    public static partial User Map(UpdateUserRequestTo updateUserRequestTo);
    public static partial User Map(CreateUserRequestTo createUserRequestTo);
    public static partial UserResponseTo Map(User User);
    public static partial IEnumerable<UserResponseTo> Map(IEnumerable<User> Users);

    public static partial IEnumerable<User> Map(
        IEnumerable<UpdateUserRequestTo> UserRequestTos);
}